File: selectionsort.java



class Solution
{
	int  select(int arr[], int i)
	{
        int min=arr[i];// code here such that selectionSort() sorts arr[]
        for(int j=i+1;j<arr.length;j++){
            if(min>arr[j]){
              min=arr[j];  
            }
           
        }
        return min;
	}
	
	void selectionSort(int arr[], int n)
	{
	    int i;
	    for (i=0;i<n;i++){
	        int min=select(arr,i);
	        for (int j=i;j<n;j++){
	            if(min==arr[j]){
	                arr[j]=arr[i];
	                arr[i]=min;
	            }
	        }
	       
	       
	    }
	   
	}
}

File: selectionSort.py
def select(self, arr, i):
        # code here
        idx = i
        for p in range(i+1,len(arr)):
            if arr[p] < arr[idx]:
                idx=p
        return idx
    
    def selectionSort(self, arr,n):
        #code here
        for current in range(n):
            min_idx=self.select(arr,current)
            arr[current],arr[min_idx]=arr[min_idx],arr[current]
